Extensions/XMPPMUCSUB/XMPPMUCSUB.m

```diff
@@ -0,0 +1,789 @@
+//
+//  XMPPMUCSUB.m
+//  Dollarama
+//
+//  Created by Besat Zardosht on 2019-05-15.
+//  Copyright © 2019 binnj. All rights reserved.
+//
+#import "XMPPMUCSUB.h"
+#import "XMPPFramework.h"
+#import "XMPPLogging.h"
+#import "XMPPIDTracker.h"
+
+#if DEBUG
+static const int xmppLogLevel = XMPP_LOG_LEVEL_VERBOSE;
+#else
+static const int xmppLogLevel = XMPP_LOG_LEVEL_WARN;
+#endif
+
+NSString *const XMPPMucSubDiscoInfo = @"http://jabber.org/protocol/disco#info";
+NSString *const XMPPMucSubErrorDomain = @"XMPPMUCSUBErrorDomain";
+NSString *const XMPPMucSubNamespace = @"urn:xmpp:mucsub:0";
+NSString *const XMPPPubSubEventNamespace = @"http://jabber.org/protocol/pubsub#event";
+NSString *const XMPPMucSubMessageNamespace = @"urn:xmpp:mucsub:nodes:messages";
+NSString *const XMPPMucSubPresenceNamespace = @"urn:xmpp:mucsub:nodes:presence";
+NSString *const XMPPMucSubSubscribeNamespace = @"urn:xmpp:mucsub:nodes:subscribe";
+NSString *const XMPPMucSubUnsubscribeNamespace = @"urn:xmpp:mucsub:nodes:unsubscribe";
+
+@interface XMPPMUCSUB()
+{
+    BOOL hasRequestedFeatures;
```

Comment:
Is there an advantage to these not being properties?

Reply:
hmm. I'm not really sure.. This is the way it's implemented in XMPPMuc and it's the main reason I didn't change it.